### DISCUSSION TITLE
"A file was included is a symlink or does not exist" when trying to overwrite `_includes/archive-single.html` using Github pages

### DISCUSSION THREAD

**Discussion by Rainymood** [mmistakes/minimal-mistakes#2977] [Q&A] [answered]
## Environment

- Minimal Mistakes version: I don't know, I run it as a remote theme
- Ruby gem or remote theme version: locally 2.6.3, remote 2.7.1
- Jekyll version: 3.9.0
- Git repository URL: https://github.com/Rainymood/rainymood.github.io
- GitHub Pages hosted (if yes provide URL to site): https://www.janmeppe.com/
- Operating system: Mac OSX

## Expected behavior

I copy over `_includes/archive-single.html` from the original minimal mistakes repository to `_includes/archive-single.html` and tweak it a bit to add the date in the title.

I can run and build it locally and the change works. Then I push it to github pages, view the commit [here](https://github.com/Rainymood/rainymood.github.io/commit/e60e5991553b69fd9095c57385335a9e9aaf242f)

## Actual behavior

Page doesn't build anymore and I get this build error:

> A file was included in /_layouts/single.html that is a symlink or does not exist in your _includes directory.

I found [this](https://stackoverflow.com/questions/22549312/github-pages-a-file-was-included-is-a-symlink-or-does-not-exist) github issue but I don't really understand what it means:

>It looks like you're using a custom source. You're not allowed to use custom sources in your Jekyll sites on GitHub Pages. They override it to your repo's top directory. So it's looking for ./_includes/... rather than ./docs/_includes.

Does this mean I can not overwrite any of the layouts? I want to tweak my minimal mistakes theme a bit, is this possible when using github pages or is this off limit?

**Comment by mmistakes** [comment, marked as answer]
Overriding `_includes` works fine. The issue is [this line](https://github.com/Rainymood/rainymood.github.io/blob/master/_includes/archive-single.html#L25) in your `_includes/archive-single.html`

```html
<!-- <a href="{{ post.url | relative_url }}" rel="permalink">{{ title }} {% include date.html %}</a> -->
```

Doesn't matter to Jekyll that you've commented it out in the HTML, you're trying to pull in `_includes/date.html` which doesn't exist in the theme nor in your local `_includes` folder.

Either rename it to `page__date.html` to use what's bundled in the theme, or create `_includes/date.html` in your repo.

**Reply by Rainymood** [reply]
That makes a lot of sense. `_includes/date.html` is still uncommitted indeed.

Thanks for clearing this up.
